Update to pass callback for nodejs4.3 runtime

briandonahue opened the pull request:
Fixes #70 - Passing along runtime to `run` method" and using a callback to handle success/fail/done scenarios.  Added deprecation warning for `context.[succeed,fail,done]()` calls.

DeviaVir commented on lib/main.js:
```diff
@@ -45,25 +45,53 @@ Lambda.prototype.run = function (program) {
     succeed: function (result) {
-      console.log('succeed: ' + JSON.stringify(result));
-      process.exit(0);
+      if(isNode43) console.log('context.succeed() is deprecated with Node.js 4.3 runtime');
```
Could you rewrite this as `if (isNode43) {` and the corresponding `}` on a newline?

DeviaVir replied:
It's also not entirely clear; it is deprecated, but can you still use it? Should we kill succeed/fail/done for anything `isNode43`?

briandonahue replied:
It is supported for backward compatibility: http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html#transition-to-new-nodejs-runtime

DeviaVir commented:
Would you mind updating https://github.com/motdotla/node-lambda/blob/master/README.md#run with this extra parameter as well? And perhaps make a note of what runtimes are currently available for both?

Greatly appreciating this work so far!

briandonahue commented:
OK, I also changed the default runtime to `nodejs4.3` since that is now the recommended version, and I added some detail to the README.  Not sure if we want to put a note in the README somewhere pointing to the docs re: the version differences?

DeviaVir commented:
@briandonahue That would be great actually, could you add a new paragraph below https://github.com/motdotla/node-lambda#custom-environment-variables ?
Will merge this when that's in!

DeviaVir commented:
Thanks @briandonahue!

briandonahue commented:
Sure thing - thanks to all the contributors, node-lambda has made playing with AWS Lambda stuff much easier!
